### PR TITLE
Embedded Single Card: get loadable time series selector

### PR DESCRIPTION
**Pull request by bhuang1323**
## Motivation for features / changes
Create a separate selector so that the single card fetch the time series data with a given card Id. 

## Technical description of changes
Separate the existing time series selector to have card metadata as input so that the single card can pass in parsed metadata. 

Googlers, please see cl/547536441 for the usage of the decomposed selector.

### CODE REVIEW

File: tensorboard/webapp/metrics/store/metrics_selectors.ts [hunk 1]
```diff
@@ -116,20 +117,30 @@ export const getCardLoadState = createSelector(
   }
 );
 
+export const getLoadableTimeSeries = (
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors.ts:120]: Can you add or rearrange existing tests for this function?

File: tensorboard/webapp/metrics/store/metrics_selectors.ts [hunk 2]
```diff
@@ -116,20 +117,30 @@ export const getCardLoadState = createSelector(
...
+  cardMetadata: CardMetadata,
+  timeSeriesData: TimeSeriesData
+) => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors.ts:123]: nit: add return type

File: tensorboard/webapp/metrics/store/metrics_selectors.ts [hunk 3]
```diff
@@ -116,20 +117,30 @@ export const getCardLoadState = createSelector(
...
+    tag,
+    sample
+  );
+  return loadable ? loadable.runToSeries : null;
+};
+
 export const getCardTimeSeries = createSelector(
   selectMetricsState,
-  (state: MetricsState, cardId: CardId): DeepReadonly<RunToSeries> | null => {
+  (state: MetricsState, cardId: CardId) => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors.ts:136]: Can we keep the return type?

File: tensorboard/webapp/metrics/store/metrics_selectors_test.ts [hunk 4]
```diff
@@ -348,7 +348,7 @@ describe('metrics selectors', () => {
     });
   });
 
-  describe('getCardTimeSeries', () => {
+  describe('getLoadableTimeSeries', () => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors_test.ts:351]: The tests in this is still using getCardTimeSeries selector. Could you make another test group (another describe) for getLoadableTimeSeries?

File: tensorboard/webapp/metrics/store/metrics_selectors_test.ts [hunk 5]
```diff
@@ -402,6 +402,35 @@ describe('metrics selectors', () => {
       );
       expect(selectors.getCardTimeSeries(state, 'card-nonexistent')).toBe(null);
     });
+
+    it('returns time series with direct metadata input', () => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors_test.ts:406]: Then in the new function, this could be just "getLoadableTimeSeries"

File: tensorboard/webapp/metrics/store/metrics_selectors_test.ts [hunk 6]
```diff
@@ -402,6 +402,35 @@ describe('metrics selectors', () => {
...
+      });
+
+      expect(
+        selectors.getLoadableTimeSeries({
+          plugin: PluginType.SCALARS,
+          tag: 'tagA',
+          runId: null,
+        })(state)
+      ).toEqual(sampleScalarRunToSeries);
+    });
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/store/metrics_selectors_test.ts:433]: could you also test when it returns null?

**Review by japie1235813 — APPROVED** [review]
